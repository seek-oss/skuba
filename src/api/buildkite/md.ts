/**
 * @internal
 */
export const md = {
  terminal: (code: string) =>
    `\`\`\`term\n${code.replace(/```/g, '\\`\\`\\`')}\n\`\`\``,
};
