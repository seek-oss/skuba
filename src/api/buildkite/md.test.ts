import { md } from './md';

describe('terminal', () => {
  it.each`
    description        | code
    ${'harmless code'} | ${'echo foo'}
    ${'1 backtick'}    | ${'`echo foo`'}
    ${'3 backticks'}   | ${'```echo foo```'}
    ${'6 backticks'}   | ${'``````'}
  `('handles $description', ({ code }) =>
    expect(md.terminal(code)).toMatchSnapshot(),
  );
});
