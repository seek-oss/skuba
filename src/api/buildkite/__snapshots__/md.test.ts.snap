// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`terminal handles 1 backtick 1`] = `
"\`\`\`term
\`echo foo\`
\`\`\`"
`;

exports[`terminal handles 3 backticks 1`] = `
"\`\`\`term
\\\`\\\`\\\`echo foo\\\`\\\`\\\`
\`\`\`"
`;

exports[`terminal handles 6 backticks 1`] = `
"\`\`\`term
\\\`\\\`\\\`\\\`\\\`\\\`
\`\`\`"
`;

exports[`terminal handles harmless code 1`] = `
"\`\`\`term
echo foo
\`\`\`"
`;
